src/app_ui/sign.rs

```diff
@@ -15,50 +15,53 @@
  *  limitations under the License.
  *****************************************************************************/
 
+use core::str::from_utf8;
+
 use crate::handlers::sign_tx::Tx;
-use crate::utils::{concatenate, to_hex_all_caps};
+use crate::utils::concatenate;
 use crate::AppSW;
-use core::str::from_utf8;
 use ledger_device_ui_sdk::bitmaps::{CROSSMARK, EYE, VALIDATE_14};
 use ledger_device_ui_sdk::ui::{Field, MultiFieldReview};
 use numtoa::NumToA;
 
+const MAX_COIN_LENGTH: usize = 10;
+
+/// Displays a transaction and returns true if user approved it.
+///
+/// This method can return [`AppSW::TxDisplayFail`] error if the coin name length is too long.
+///
+/// # Arguments
+///
+/// * `tx` - Transaction to be displayed for validation
 pub fn ui_display_tx(tx: &Tx) -> Result<bool, AppSW> {
-    // Format amount value
-    let mut amount_buf = [0u8; 20];
-    let mut amount_with_denom_buf = [0u8; 25];
-    concatenate(
-        &["CRAB", " ", tx.value.numtoa_str(10, &mut amount_buf)],
-        &mut amount_with_denom_buf,
-    );
-    let amount_str_with_denom = from_utf8(&amount_with_denom_buf)
-        .map_err(|_| AppSW::TxDisplayFail)?
-        .trim_matches(char::from(0));
+    // Generate string for amount
+    let mut numtoa_buf = [0u8; 20];
+    let mut value_buf = [0u8; 20 + MAX_COIN_LENGTH + 1];
 
-    // Format destination address
-    let hex_addr_buf = to_hex_all_caps(&tx.to).map_err(|_| AppSW::TxDisplayFail)?;
-    let hex_addr_str = from_utf8(&hex_addr_buf).map_err(|_| AppSW::TxDisplayFail)?;
-    let mut addr_with_prefix_buf = [0u8; 42];
-    concatenate(&["0x", hex_addr_str], &mut addr_with_prefix_buf);
-    let hex_addr_str_with_prefix =
-        from_utf8(&addr_with_prefix_buf).map_err(|_| AppSW::TxDisplayFail)?;
+    let value_str = concatenate(
+        &[tx.coin, &" ", tx.value.numtoa_str(10, &mut numtoa_buf)],
+        &mut value_buf,
+    )
+    .map_err(|_| AppSW::TxDisplayFail)?; // Fails if value_buf is too small
 
-    // Format memo
-    let memo_str = from_utf8(&tx.memo[..tx.memo_len as usize]).map_err(|_| AppSW::TxDisplayFail)?;
+    // Generate destination address string in hexadecimal format.
+    let mut to_str = [0u8; 42];
+    to_str[..2].copy_from_slice("0x".as_bytes());
+    hex::encode_to_slice(tx.to, &mut to_str[2..]).unwrap();
 
     // Define transaction review fields
     let my_fields = [
         Field {
             name: "Amount",
-            value: amount_str_with_denom,
+            value: value_str,
         },
         Field {
             name: "Destination",
-            value: hex_addr_str_with_prefix,
+            value: core::str::from_utf8(&to_str).unwrap(),
```

Comment:
@kingofpayne we should probably display the address capitalized like for the `get pub key` command ?

Reply:
Change done.